docs/pai-protocol.yaml

```diff
@@ -0,0 +1,217 @@
+protocolVersion: String, required # Protocol version, current version is 2
+name: String, required
+type: String, required # The type of the component. Must be one of the following: job, data, script, dockerimage, or output
+version: String, optional # Component version. Default is latest
+contributor: String, optional
+description: String, optional
+
+prerequisites: # Optional
+  - protocolVersion: String, optional # If omitted, follow the protocolVersion in root
+    name: String, required
+    type: String, required # Component type. Must be one of the following: data, script, dockerimage, or output. Prerequisites.type cannot be "job"
+    version: String, optional # Component version. Default is latest
+    contributor: String, optional
+    description: String, optional
+    auth: Object, optional # Only available when the type is dockerimage
+      username: String, optional
+      passward: String, optional
+      registryuri: String, optional
+    uri: String or list, required # Only when the type is data can the uri be a list
+
+parameters: # Optional, can be omitted. If specified, the whole object can be referenced as `$parameters`
+  <param1>: value1Type # Specify name and value of all the referencable parameters that will be used in the whole job template.
+  <param2>: value2Type # Can be referenced by `<% $parameters.param1 %>`, `<% $parameters.param2 %>`
+
+jobRetryCount: Integer, optional # Default is 0
+taskRoles:
+  <name>: String, required  # Name of the taskRole
+   protocol_version: String, optional # Protocol version, default is 2
+    instances: Integer, optional # Default is 1, instances of a taskRole
+    completion:
```

Comment:
please explain what is "completion". e.g., Completion specifies how a job state changes in case of a task completion. 
Please provide a link to our document for details.

Reply:
Updated.